lib/_http_parser.js

```diff
@@ -0,0 +1,907 @@
+          // multiple spaces/tabs separating the URL and HTTP version
+          var ch = buf.charCodeAt(end + 1);
+          if (ch === 49)
+            minor = 1;
+          else if (ch === 48)
+            minor = 0;
+          else
+            return this._setError('Malformed request line');
+          var h = 0;
+          while (end >= 0) {
+            var ch = buf.charCodeAt(end);
+            if (h < 8) {
+              if (ch !== REQ_HTTP_VER_BYTES[h++])
+                return this._setError('Malformed request line');
+            } else if (ch >= 33 && ch !== 127) {
+              urlEnd = end + 1;
+              break;
+            }
+            --end;
+          }
+          if (urlEnd === undefined)
+            return this._setError('Malformed request line');
+
+          // Now start working forwards and both validate the HTTP method and
+          // find the start of the URL (in case there are multiple spaces/tabs
+          // separating the method and the URL
+          var isConnect = false;
+          var c = 0;
+          for (var i = 0; i < urlEnd; ++i) {
+            ch = buf.charCodeAt(i);
+            if (firstSP !== undefined) {
+              if (ch >= 33 && ch !== 127) {
+                urlStart = i;
+                break;
+              }
+            } else if (ch === 32) {
+              firstSP = i;
+              isConnect = (c === 7);
+            } else if (ch < 33 || ch > 126)
+              return this._setError('Malformed request line');
+            else if (c >= 0) {
+              if (c === 7)
+                c = -1;
+              else {
+                ch |= 0x20;
+                if (ch !== CC_CONNECT[c])
+                  c = -1;
+                else
+                  ++c;
+              }
+            }
+          }
+          if (firstSP === undefined ||
+              urlStart === undefined ||
+              urlStart === urlEnd) {
+            return this._setError('Malformed request line');
+          }
+
+          var url = buf.slice(urlStart, urlEnd);
+          if (!validateUrl(url, isConnect))
+            return this._setError('Malformed request line');
+
+          if (isConnect)
+            this._flags |= FLAG_CONNECT_METHOD;
+          this.httpMinor = minor;
+          this.method = buf.slice(0, firstSP);
+          this.url = url;
+
+          state = STATE_HEADER;
+          break;
+        case STATE_STATUS_LINE:
+          // Original HTTP parser ignored blank lines before request/status
+          // line, so we do that here too ...
+          if (buflen === 0 || buf.charCodeAt(0) === CR)
+            break;
+
+          // Validate HTTP version
+          for (var h = 0; i < 7; ++h) {
+            if (buf.charCodeAt(i) !== RES_HTTP_VER_BYTES[h])
+              return this._setError('Malformed status line');
+          }
+          var minor;
+          var status = 0;
+          if (buf.charCodeAt(7) === 49)
+            minor = 1;
```

Comment:
`minor` is only used once in this case, and in the case below. I think it would improve readability and maybe? but probably not performance to just directly set `this.httpMinor`.